capa/ghidra/capa_explorer.py

```diff
@@ -359,9 +363,26 @@ def main():
         popup("capa explorer found no matches.")  # type: ignore [name-defined] # noqa: F821
         return capa.main.E_EMPTY_REPORT
 
-    for item in parse_json(capa_data):
-        item.bookmark_functions()
-        item.label_matches()
+    user_choice = askChoice(  # type: ignore [name-defined] # noqa: F821
+        "Choose b/w bookmarks & comments",
+        "preferred action:",
+        ["bookmarks", "comments", "both", "none"],
+        "both",
+    )
+
+    if user_choice == "bookmarks":
+        for item in parse_json(capa_data):
+            item.bookmark_functions()
+    elif user_choice == "comments":
+        for item in parse_json(capa_data):
+            item.label_matches()
+    elif user_choice == "both":
+        for item in parse_json(capa_data):
+            item.bookmark_functions()
+            item.label_matches()
+    else:
+        pass
```

Comment:
This script modifies a Ghidra database by adding:
1. new namespace named "capa" and corresponding namespace entries
2. pre/plate-comments
3. function bookmarks

We want our changes here to enable users to select any number of these options when running this script. Presently, the `label_matches` method creates the "capa" namespace, corresponding namespace entries, and pre/plate-comments. We need to modify the `label_matches` method to account for the user's selection and I'd recommend passing new boolean arguments to the `label_matches` method to implement this. Creating a new "capa" namespace and corresponding namespace entries should be grouped as one option, likewise with setting pre/plate-comments.

Reply:
@mike-hunhoff 
So just to clarify for the adding comments based on the user input, we want to just have a boolean default and pass an argument based on what the user wants in the function call within the if-else statement.

Further, if I understand right you'd like separate options for creating the 'capa' namespace and setting pre/plate comments, even though the namespace might be used for comments. Is that correct?

Can you elaborate on the rationale behind separate options? Is it for user flexibility or potential workflow reasons?

Reply:
Taking a step back, it may be easier to understand the requested changes by reviewing the [README's UI integration section](https://github.com/mandiant/capa/tree/master/capa/ghidra#ui-integration).

capa_explorer.py does three things:
1. Add Ghidra top-level namespace named "capa" that is viewable in Ghidra's Symbol Tree window
2. Add pre/plate comments that are viewable in Ghidra's Disassembly Listing and Decompiler windows
3. Add bookmarks that are viewable in Ghidra's Bookmarks window

We'd like to give user's an option to choose which of these three things are executed based on their needs/workflow. For example, a user may be interested in options 1 and 3 while not wanting 100s of comments added to their Ghidra database by option 2.